pallets/tokens/src/lib.rs

```diff
@@ -875,3 +911,256 @@ where
+				} else {
+					SignedImbalance::Negative(MultiTokenNegativeImbalance::new(currency_id, account.free - value))
+				};
+				account.free = value;
+				Ok(imbalance)
+			},
+		)
+		.unwrap_or_else(|_| SignedImbalance::Positive(MultiTokenPositiveImbalance::zero(currency_id)))
+	}
+}
+
+impl<T> MultiTokenReservableCurrency<T::AccountId> for MultiTokenCurrencyAdapter<T>
+where
+	T: Trait,
+{
+	fn can_reserve(currency_id : T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> bool {
+		Module::<T>::can_reserve(currency_id, who, value)
+	}
+
+	fn slash_reserved(currency_id: T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> (Self::NegativeImbalance, Self::Balance) {
+		let actual = Module::<T>::slash_reserved(currency_id, who, value);
+		(MultiTokenNegativeImbalance::zero(currency_id), actual)
+	}
+
+	fn reserved_balance(currency_id: T::CurrencyId, who: &T::AccountId) -> Self::Balance {
+		Module::<T>::reserved_balance(currency_id, who)
+	}
+
+	fn reserve(currency_id: T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> DispatchResult {
+		Module::<T>::reserve(currency_id, who, value)
+	}
+
+	fn unreserve(currency_id: T::CurrencyId, who: &T::AccountId, value: Self::Balance) -> Self::Balance {
+		Module::<T>::unreserve(currency_id, who, value)
+	}
+
+	fn repatriate_reserved(
+        currency_id: T::CurrencyId,
+		slashed: &T::AccountId,
+		beneficiary: &T::AccountId,
+		value: Self::Balance,
+		status: Status,
+	) -> result::Result<Self::Balance, DispatchError> {
+		Module::<T>::repatriate_reserved(currency_id, slashed, beneficiary, value, status)
+	}
+}
+
+impl<T> MultiTokenLockableCurrency<T::AccountId> for MultiTokenCurrencyAdapter<T>
+where
+	T: Trait,
+{
+	type Moment = T::BlockNumber;
+	type MaxLocks = ();
+
+	fn set_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId, amount: Self::Balance, _reasons: WithdrawReasons) {
+		Module::<T>::set_lock(id, currency_id, who, amount)
+	}
+
+	fn extend_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId, amount: Self::Balance, _reasons: WithdrawReasons) {
+		Module::<T>::extend_lock(id, currency_id, who, amount)
+	}
+
+	fn remove_lock(currency_id: T::CurrencyId, id: LockIdentifier, who: &T::AccountId) {
+		Module::<T>::remove_lock(id, currency_id, who)
+	}
+}
+
+
+impl<T> MultiTokenCreatableCurrency<T::AccountId> for MultiTokenCurrencyAdapter<T> where
+    T: Trait
+{
+	fn create(address: &T::AccountId, amount: T::Balance) -> T::CurrencyId{
+        let token_id = <NextCurrencyId<T>>::get();
+		<NextCurrencyId<T>>::mutate(|id| *id += One::one());
+        let tokens = <Self as MultiTokenCurrency<T::AccountId>>::issue(token_id, amount);
+        // we are creating new token so amount can not be overflowed as its always true
+        // 0 + amount < T::Balance::max_value()
+        let _ = <Self as MultiTokenCurrency<T::AccountId>>::resolve_into_existing(token_id, address, tokens);
```

Comment:
shouldn't this be resolve_creating instead?

Reply:
Even though I believe they are both are functionally the same as far as the currency ORML tokens are concerned.

Reply:
eventualy i used deposit_creating that works with `Balance` instead of `Imbalance`